### PR TITLE
switch hyperkube to cobra

### PR DESCRIPTION
**Pull request by deads2k**
This makes use of cobra to layer the commands with a library that already supports composing commands.

Fixes https://github.com/kubernetes/kubernetes/issues/34732

```release-note
NONE
```

### CODE REVIEW

**Comment by deads2k** [comment]
@ncdc

File: cmd/hyperkube/server.go [hunk 1]
```diff
@@ -1,77 +0,0 @@
-/*
```
Comment (smarterclayton) [inline cmd/hyperkube/server.go]: Very satisfying

**Comment by smarterclayton** [comment]
Looks fine to me, but waiting for the remaining prereq PR to finalize comments

File: hack/local-up-cluster.sh [hunk 2]
```diff
@@ -539,7 +539,7 @@ function start_apiserver {
     fi
 
     APISERVER_LOG=${LOG_DIR}/kube-apiserver.log
-    ${CONTROLPLANE_SUDO} "${GO_OUT}/hyperkube" apiserver ${swagger_arg} ${audit_arg} ${authorizer_arg} ${priv_arg} ${runtime_config} \
+    ${CONTROLPLANE_SUDO} "${GO_OUT}/hyperkube" kube-apiserver ${swagger_arg} ${audit_arg} ${authorizer_arg} ${priv_arg} ${runtime_config} \
```
Comment (smarterclayton) [inline hack/local-up-cluster.sh]: ? Isn't this a breaking change?  Or is the old one deprecated?

Comment (deads2k) [reply hack/local-up-cluster.sh]: > ? Isn't this a breaking change? Or is the old one deprecated?

Gotta be a bug from 1.0, right?  I can re-alias it if need be.

**Comment by deads2k** [comment]
@smarterclayton aliased the commands

File: cmd/hyperkube/hyperkube.go [hunk 3]
```diff
@@ -18,259 +18,43 @@ package main
...
-{{.Long | trim | wrap "    "}}{{end}}
-Call '{{.Name}} --make-symlinks' to create symlinks for each server in the local directory.
-Call '{{.Name}} <server> --help' for help on a specific server.
-`
-	utiltemplate.ExecuteTemplate(hk.Out(), tt, hk)
-}
-
-// MakeSymlinks will create a symlink for each registered hyperkube server in the local directory.
-func (hk *HyperKube) MakeSymlinks(command string) error {
+//func Usage() {
```
[the remark this answers is not in the document]
Comment (deads2k) [reply cmd/hyperkube/hyperkube.go]: > is there a reason to keep the usage template as a comment?

I'm happy to remove it.

**Comment by deads2k** [comment]
No with symlinks and different behavior based on binary name.

**Comment by deads2k** [comment]
@smarterclayton let's tie this off

**Comment by deads2k** [comment]
> /lgtm

You're up early :)

**Comment by deads2k** [comment]
/approve

to kick bot

**Pull request merged**